be/src/util/faststring.cc

```diff
@@ -37,18 +37,20 @@ void faststring::GrowToAtLeast(size_t newcapacity) {
 
 void faststring::GrowArray(size_t newcapacity) {
     DCHECK_GE(newcapacity, capacity_);
-    std::unique_ptr<uint8_t[]> newdata(new uint8_t[newcapacity]);
-    if (len_ > 0) {
-        memcpy(&newdata[0], &data_[0], len_);
-    }
-    capacity_ = newcapacity;
+    uint8_t* newdata = NULL;
     if (data_ != initial_data_) {
-        delete[] data_;
+        newdata = (uint8_t*)realloc((void*)data_, newcapacity);
     } else {
+        newdata = (uint8_t*)malloc(newcapacity);
+        if (len_ > 0) {
+            memcpy(newdata, data_, len_);
+        }
+    }
+    capacity_ = newcapacity;
+    data_ = newdata;
+    if (data_ == initial_data_) {
         ASAN_POISON_MEMORY_REGION(initial_data_, arraysize(initial_data_));
```

Comment:
data_ never equal to initial_data_?

Reply:
initial_data_ is a reserved memory, which is no longer equal to after the first GrowArray.

Reply:
yep, there is a bug visible to the naked eye.